nixos/modules/services/misc/klipper.nix

```
@@ -118,14 +160,36 @@ in
       let
         klippyArgs = "--input-tty=${cfg.inputTTY}"
           + optionalString (cfg.apiSocket != null) " --api-server=${cfg.apiSocket}";
+        printerConfigPath =
+          if cfg.isConfigMutable
+          then cfg.mutableConfigPath
+          else "/etc/klipper.cfg";
+        printerConfigFile =
+          if cfg.settings != null
+          then format.generate "klipper.cfg" cfg.settings
+          else cfg.configFile;
       in
       {
         description = "Klipper 3D Printer Firmware";
         wantedBy = [ "multi-user.target" ];
         after = [ "network.target" ];
 
         serviceConfig = {
-          ExecStart = "${cfg.package}/lib/klipper/klippy.py ${klippyArgs} /etc/klipper.cfg";
+          ExecStartPre = pkgs.writeScript "init-klipper" ''
+            ${lib.optionalString (cfg.isConfigMutable) ''
+              [ -e ${printerConfigPath} ] || {
+                mkdir -p $(dirname ${printerConfigPath})
+                cp ${printerConfigFile} ${printerConfigPath}
+                chmod +w ${printerConfigPath}
+              }
+            ''}
+            ${lib.optionalString (cfg.gcodeFolder != null) ''
+              [ -e ${cfg.gcodeFolder} ] || {
+                mkdir -p ${cfg.gcodeFolder}
+              }
```

```suggestion
              mkdir -p ${cfg.gcodeFolder}
```
``mkdir -p`` should be a noop when the directory exists.